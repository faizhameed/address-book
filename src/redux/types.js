export const actionTypes = {
  ADD_BOOK: "ADD_BOOK",
};
