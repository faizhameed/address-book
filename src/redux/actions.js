import { actionTypes } from "./types";
import Axios from "axios";

export const fetchContactList = (page = 1) => {
  let url = `https://randomuser.me/api/?${page}&results=50`;

  return function (dispatch) {
    dispatch({
      type: actionTypes.GET_CONTACTS_PENDING,
    });
    Axios.get(url)
      .then((response) =>
        dispatch({
          type: actionTypes.GET_CONTACTS_SUCCESS,
          payload: response.data,
        })
      )
      .catch((response) =>
        dispatch({
          type: actionTypes.GET_CONTACTS_FAILED,
          payload: response.error,
        })
      );
  };
};

export const updatePageNumber = (page) => ({
  type: actionTypes.UPDATE_PAGE_NUMBER,
  payload: page,
});

export const searchFieldChange = (data) => ({
  type: actionTypes.SEARCH_FIELD_CHANGE,
  payload: data,
});

export const updatePersonDetailView = (person) => ({
  type: actionTypes.UPDATE_PERSON_DETAIL,
  payload: person,
});

export const toggleDarkMode = () => ({
  type: actionTypes.TOGGLE_DARK_MODE,
});
