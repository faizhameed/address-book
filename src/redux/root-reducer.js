import { combineReducers } from "redux";
import { actionTypes } from "./types";

const initialStateContactList = {
  contactList: [],
  error: null,
  isPending: false,
  page: 1,
};
function contactListReducer(state = initialStateContactList, action) {
  switch (action.type) {
    case actionTypes.GET_CONTACTS_SUCCESS:
      return {
        ...state,
        contactList: [...state.contactList, ...action.payload.results],
        isPending: false,
      };

    case actionTypes.GET_CONTACTS_PENDING:
      return {
        ...state,
        isPending: true,
      };
    case actionTypes.GET_CONTACTS_FAILED:
      return {
        ...state,
        isPending: false,
        error: action.payload,
      };

    case actionTypes.UPDATE_PAGE_NUMBER:
      if (action.payload < 11) {
        return {
          ...state,
          page: action.payload,
        };
      } else {
        return state;
      }
    default:
      break;
  }
  return state;
}

const rootReducer = combineReducers({
  contactListReducer,
});

export default rootReducer;
