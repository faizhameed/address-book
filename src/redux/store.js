import { createStore } from "redux";
import rootReducer from "./root-reducer";
import { applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import thunkMiddleware from "redux-thunk";

const logger = createLogger();
export const store = createStore(
  rootReducer,
  applyMiddleware(thunkMiddleware, logger)
);
