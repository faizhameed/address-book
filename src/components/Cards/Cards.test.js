import { shallow } from "enzyme";
import ProfileCard from "./Card";
import React from "react";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

describe("Profile Card Component ", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <ProfileCard />
      </Provider>
    );
  });

  it("should render profileCard component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
