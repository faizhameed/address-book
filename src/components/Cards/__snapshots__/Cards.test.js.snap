// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Card Component  should render profileCard component 1`] = `ShallowWrapper {}`;
